common/presenters/move-to-message-banner-component.js

```diff
@@ -0,0 +1,147 @@
+const { isEmpty, kebabCase } = require('lodash')
+
+const i18n = require('../../config/i18n')
+const filters = require('../../config/nunjucks/filters')
+const componentService = require('../services/component')
+
+const frameworkToTaskListComponent = require('./framework-to-task-list-component')
+
+function _returnComponent(title, content) {
+  return {
+    allowDismiss: false,
+    classes: 'app-message--instruction govuk-!-padding-right-0',
+    title: {
+      text: title,
+    },
+    content: {
+      html: content,
+    },
+  }
+}
+
+function moveToMessageBannerComponent({
+  move,
+  moveUrl,
+  canAccess,
+  assessmentType,
+}) {
+  const { profile, status } = move
+  const { id: profileId, [assessmentType]: assessment } = profile || {}
+  const baseUrl = `${moveUrl}/${kebabCase(assessmentType)}`
+  const context = assessmentType
+
+  if (['proposed'].includes(status) || !profileId) {
+    return undefined
+  }
+
+  let title
+  let content
+
+  // No PER exists
+  if (['requested', 'booked'].includes(status) && isEmpty(assessment)) {
+    title = i18n.t('messages::assessment.pending.heading', { context })
+    content = `<p>${i18n.t('messages::assessment.pending.content', {
+      context,
+    })}</p>`
+
+    if (canAccess(`${assessmentType}:create`)) {
+      content += `
+        ${componentService.getComponent('govukButton', {
+          href: `${baseUrl}/new`,
+          text: i18n.t('actions::start_assessment', {
+            context,
+          }),
+        })}
+      `
+    }
+
+    return _returnComponent(title, content)
+  }
+
+  const {
+    _framework = {},
+    meta = {},
+    status: perStatus,
+    confirmed_at: confirmedAt,
+  } = assessment || {}
+
+  // PER has been confirmed
+  if (assessment && perStatus === 'confirmed') {
```

Comment:
This block and the one below both check for `assessment`, but we've already partially checked to see if this is empty beforehand.

Perhaps this existence check could be done first, and then an appropriate action taken instead? This might make the logic clearer as well.